### ISSUE TITLE
RDP Use client kbd mapping doesn't pick up AltGr + key combos

### ISSUE THREAD

**Issue by jpeg729** [FreeRDP/Remmina#1338]
Running v1.2.0-rcgit.23 on Ubuntu 14.04 Unity connecting to Windows Server 2008.

The `Use client keyboard mapping` option is useless to me because it doesn't pick up `AltGr + key` combos. Nothing happens when I use such a combo, nothing gets typed.

This doesn't just affect me with my customised keyboard layout, it affects anyone trying to use a french keyboard layout. To type `#` on a french keyboard you need to use the `AltGr + 3` combo.

I am sure there are plenty of other keyboard layouts affected by this issue.

**Comment by antenore** [comment]
I've a French and a Swiss French keyboard and it works for me.

The only difference is that I've a quite newer distribution. @giox069 any ideas?

**Comment by antenore** [comment]
Ah! One more thing, be sure to have Grab all keys activated (a keyboard icon in the Remmina toolbar)

**Comment by jpeg729** [comment]
Grab all keys doesn't make the slightest bit of difference.

**Comment by jpeg729** [comment]
Another annoying thing is that `Ctrl-key` combos don't use my local keyboard layout.

**Comment by antenore** [comment]
Crap... I don't have any clue at the moment.

Two things can you test with xfreerdp from the command line (see xfreerdp --help for the right options)? Do you hve a more updated Linux dstribution to test?

Merci!

**Comment by jpeg729** [comment]
I don't have a more up-to-date linux distro on hand.

I can test with xfreerdp. I have the version `2.0.0~git201710191115-0+remmina201710192147.r86bb1e0.d0f49046~ubuntu14.04.1` which presumably comes from the remmina ppa.
What options could I usefully test?

**Comment by antenore** [comment]
```
xfreerdp +clipboard /kbd: 0x0000040C /v:hostname /u:username
```

Try also without the ```/kbd``` option

**Comment by giox069** [comment]
altgr+key should work, are you using a standard altgr+key or a customized/mapped one?
ctrl+key will never map, read comments here: https://github.com/FreeRDP/Remmina/blob/next/remmina-plugins/rdp/rdp_event.c#L656
Keyboard mapping feature will never work as the users expects. I think we should remove it. Or implement 2 different types of client keyboard mapping (hwcode->hwcode and hwcode->unicode, but both will be incomplete). Or warning the user that he never will get what he/she thinks to get.

**Comment by jpeg729** [comment]
Unless I use the `Use client keyboard mapping` option in remmina I always get the US keyboard layout.

I have tried asking Remmina to use a specific layout, and I have tried with and without the `/kbd` option in xfreerdp, but I always get the US layout, which is annoying but I have gotten used to it by now.

**Comment by tsdh** [comment]
@giox069 I agree that client side keyboard mapping is hell, and I suffer from Ctrl+C/V  actually mean Ctrl+<the keys that are C/V in US layout>, but please never ever remove it!  As a user of the German Dvorak Type II keyboard layout (which is not available on stock Windows), I wouldn't be able to use remmina at all.

BTW: the client mapping provided by the rdesktop cmd line client works also for modified keys but rdesktop doesn't work with never authentication methods...

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @giox069 @jpeg729 @tsdh
